stempy/image.cpp

```diff
@@ -338,17 +338,15 @@ std::vector<int> createSTEMHistogram(const STEMImage& inImage,
   return frequencies;
 }
 
-vector<uint16_t> expandSparsifiedData(const vector<vector<uint32_t>>& data,
-                                      size_t numPixels)
+void calculateSTEMValuesSparse(const vector<vector<uint32_t>>& data,
+                               uint16_t* mask, STEMImage& image)
 {
-  vector<uint16_t> ret(data.size() * numPixels, 0);
-  for (size_t i = 0; i < data.size(); ++i) {
-    for (auto pos : data[i]) {
-      ret[i * numPixels + pos] = 1;
-    }
+  for (unsigned i = 0; i < data.size(); ++i) {
+    uint64_t values = 0;
+    for (auto pos : data[i])
+      values += mask[pos];
```

Comment:
Lets use `{ }` for the for loop, even if there is one line at the moment. Is so easy to add another line without thinking and it not part of the loop!

Reply:
Done!